htdocs/core/lib/modulebuilder.lib.php

```diff
@@ -321,7 +321,7 @@ function rebuildObjectSql($destdir, $module, $objectname, $newmask, $readdir = '
             	$tmp = explode('.', $val['foreignkey']);
             	if (!empty($tmp[0]) && !empty($tmp[1]))
             	{
-            		$texttoinsert .= "ALTER TABLE llx_".strtolower($module).'_'.strtolower($objectname)." ADD CONSTRAINT llx_".strtolower($module).'_'.strtolower($objectname)."_".$key." FOREIGN KEY (".$key.") REFERENCES ".$tmp[0]."(".$tmp[1].");";
+            		$texttoinsert .= "ALTER TABLE llx_".strtolower($module).'_'.strtolower($objectname)." ADD CONSTRAINT llx_".strtolower($module).'_'.strtolower($objectname)."_".$key." FOREIGN KEY (".$key.") REFERENCES ".MAIN_DB_PREFIX.$tmp[0]."(".$tmp[1].");";
```

Comment:
in update files MAIN_DB_PREFIX is always llx_

Reply:
For sql files, we always use llx_ as prefix (because we need a value common to everybody). It is the upgrade or loading function that will replace llx_ with the MAIN_DP_PREFIX of the user.